File: LocationAPI/UpdateLocation.py
import requests
import json

locationId = "LOCATIONID OF LOCATION TO BE UPDATED"
locationName = "My Location Name"
customId = "my-custom-location-id"
products = ["nucd"] # intelligence products the location will be subscribed to

data = {
  "properties": {
    "locationName": locationName,
    "customId": customId,
    "products": products,
  }
}

url = ("https://location-api.hibirdi.com/locations/{}"
       .format(locationId))
headers = {"Content-Type": "application/json"}

response = requests.request("PUT", url, data=(json.dumps(data)), headers=headers)

print(response.text)

File: LocationAPI/UpdateLocation.js
const xhr = new XMLHttpRequest();

const locationId = "LOCATIONID OF LOCATION TO BE UPDATED";
const locationName = "My Location Name";
const customId = "my-custom-location-id";
const products = ["nucd"]; // intelligence products the location will be subscribed to

const data = {
  "properties": {
    "locationName": locationName,
    "customId": customId,
    "products": products,
  }
}
const url = `https://location-api.hibirdi.com/locations/${locationId}`;

xhr.addEventListener("readystatechange", function () {
  if (this.readyState === 4) {
    console.log(this.responseText);
  }
});

xhr.open("PUT", url);
xhr.setRequestHeader('Content-Type', 'application/json');
xhr.send(JSON.stringify(data));
